packages/console/src/ConsoleStatusBar.tsx

```diff
@@ -1,136 +1,110 @@
-  };
-
-  constructor(props: ConsoleStatusBarProps) {
-    super(props);
-
-    this.handleCommandStarted = this.handleCommandStarted.bind(this);
-    this.handleCommandCompleted = this.handleCommandCompleted.bind(this);
-
-    this.pending = new Pending();
-
-    this.state = {
-      isDisconnected: false,
-      isCommandRunning: false,
-    };
-  }
-
-  componentDidMount(): void {
-    this.startListening();
-  }
+function ConsoleStatusBar({
+  children,
+  dh,
+  session,
+  overflowActions,
+}: ConsoleStatusBarProps): ReactElement {
+  const [isCommandRunning, setIsCommandRunning] = useState(false);
 
-  componentWillUnmount(): void {
-    this.stopListening();
-    this.cancelPendingPromises();
-  }
-
-  pending: Pending;
-
-  startListening(): void {
-    const { dh, session } = this.props;
-    session.addEventListener(
-      dh.IdeSession.EVENT_COMMANDSTARTED,
-      this.handleCommandStarted
-    );
-  }
+  const handleCommandStarted = useCallback(async (event: CustomEvent) => {
+    setIsCommandRunning(true);
 
-  stopListening(): void {
-    const { dh, session } = this.props;
-    session.removeEventListener(
-      dh.IdeSession.EVENT_COMMANDSTARTED,
-      this.handleCommandStarted
-    );
-  }
-
-  cancelPendingPromises(): void {
-    this.pending.cancel();
-  }
-
-  handleCommandStarted(event: CustomEvent): void {
     const { result } = event.detail;
-
-    this.pending
-      .add(result)
-      .then(() => this.handleCommandCompleted(null))
-      .catch(error => this.handleCommandCompleted(error));
-
-    this.setState({
-      isCommandRunning: true,
-    });
-  }
-
-  handleCommandCompleted(error: unknown): void {
-    // Don't update state if the promise was canceled
-    if (!(error instanceof CanceledPromiseError)) {
-      this.setState({
-        isCommandRunning: this.pending.pending.length > 0,
-      });
+    try {
+      await result;
+    } catch (error) {
+      // No-op, fall through
     }
-  }
-
-  render(): ReactElement {
-    const { children, dh, openObject, overflowActions, objects } = this.props;
-    const { isDisconnected, isCommandRunning } = this.state;
 
-    let statusIconClass = null;
-    let tooltipText = null;
+    setIsCommandRunning(false);
+  }, []);
+
+  useEffect(
+    function startListening() {
+      session.addEventListener(
+        dh.IdeSession.EVENT_COMMANDSTARTED,
+        handleCommandStarted
+      );
+
+      return function stopListening() {
+        session.removeEventListener(
+          dh.IdeSession.EVENT_COMMANDSTARTED,
+          handleCommandStarted
+        );
+      };
+    },
+    [dh, handleCommandStarted, session]
+  );
+
+  let statusIconClass = null;
+  let tooltipText = null;
+  if (isCommandRunning) {
+    // Connected, Pending
+    statusIconClass = 'console-status-icon-pending';
+    tooltipText = 'Worker is busy';
+  } else {
+    // Connected, Idle
+    statusIconClass = 'console-status-icon-idle';
+    tooltipText = 'Worker is idle';
+  }
```

Comment:
No disconnected state after the refactor? This component is consumed by DHE, but I'm not sure how to test/get the disconnected state there.

Reply:
Actually nevermind, `isDisconnected` isn't used at all in this component. It's always `false`, and it never gets set anywhere. That's why I removed it, it was never used.